snow/engine/snowman/transitive.go

```diff
@@ -453,7 +474,18 @@ func (t *Transitive) issueFrom(vdr ids.ShortID, blk snowman.Block) (bool, error)
 			return false, nil
 		}
 	}
-	return t.Consensus.Issued(blk), nil
+
+	// Remove any outstanding requests for this block
+	t.blkReqs.RemoveAny(blkID)
+
+	issued := t.Consensus.Issued(blk)
+	if issued {
```

Comment:
Perhaps not right now but let's add a comment for this because idt it's super intuitive